src/main/java/org/terasology/logic/health/DamageAuthoritySystem.java

```diff
@@ -236,4 +236,18 @@ public void onCrash(HorizontalCollisionEvent event, EntityRef entity, CharacterS
             }
         }
     }
+    /**
+     * Reduces the baseDamage of BeforeDamagedEvent if DamageResistComponent is added.
+     *
+     * @param event BeforeDamagedEvent sent before inflicting damage
+     * @param entity Entity which suffered some type of damage
+     */
+    @ReceiveEvent
+    public void onResistMode(BeforeDamagedEvent event, EntityRef entity, DamageResistComponent resistanceComponent) {
+        String damageType=event.getDamageType().getName();
+        String subString=damageType.substring(7);
```

Comment:
Be wary of using substring with a hard coded number - the value may change or leave a viewer wondering what it means ("magic numbers") - could lead to bugs later :-)

You may want to:

* Find a way to dynamically find the right spot (like using RegEx)
* Alternatively extract the number into a namked constant to reduce confusion
* Harden the code in case if a failure to find expected values (look for an index of a desired char, if not found then log an error / throw an exception, etc)